agent/structs/connect_ca.go

```diff
@@ -67,9 +67,14 @@ type CARoot struct {
 	// SerialNumber is the x509 serial number of the certificate.
 	SerialNumber uint64
 
-	// SigningKeyID is the ID of the public key that corresponds to the private
-	// key used to sign leaf certificates. Is is the HexString format of the
-	// raw AuthorityKeyID bytes.
+	// SigningKeyID is the connect.HexString encoded id of the public key that
+	// corresponds to the private key used to sign leaf certificates in the
+	// local datacenter.
+	//
+	// The value comes from x509.Certificate.SubjectKeyId of the local leaf
+	// signing cert.
+	//
+	// See https://www.rfc-editor.org/rfc/rfc3280#section-4.2.1.1 for more detail.
```

Comment:
Calling out this docs change because it is significant. Previously we documented this as coming from the `AuthorityKeyId`. 

From my reading of the linked x509 RFC, it sounds like these two values (AuthorityKeyId and SubjectKeyId) are the same on a self-signed root cert, but otherwise are different.

On non-root certs the AuthorityKeyID is the `parentCert.SubjectKeyId`, and the SubjectKeyId is the encoded ID of the public key.

So I believe this field is the `SubjectKeyId` of the local signing cert (**not** the `AuthorityKeyID` of that cert). My reading of the code seems to match this understanding. We generally set this to `intermediateCert.SubjectKeyId`. The exception is in `parseCARoot`, which I believe is a bug, and I've documented that in this issue: #11662